File: backend/src/Allegro.Domain/Dome.cs
namespace Allegro.Domain;

/// <summary>
/// Un domo del glamping. Inicialmente existen dos, creados mediante datos iniciales.
/// </summary>
public class Dome
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public string Name { get; set; } = string.Empty;

    public string ShortDescription { get; set; } = string.Empty;

    /// <summary>Capacidad máxima de huéspedes.</summary>
    public int MaxCapacity { get; set; }

    public bool IsActive { get; set; } = true;

    public ICollection<Reservation> Reservations { get; set; } = new List<Reservation>();
}

File: mobile/lib/src/models/dome.dart
class Dome {
  final String id;
  final String name;
  final String shortDescription;
  final int maxCapacity;
  final bool isActive;

  const Dome({
    required this.id,
    required this.name,
    required this.shortDescription,
    required this.maxCapacity,
    required this.isActive,
  });

  factory Dome.fromJson(Map<String, dynamic> json) => Dome(
    id: json['id'] as String,
    name: json['name'] as String,
    shortDescription: (json['shortDescription'] as String?) ?? '',
    maxCapacity: json['maxCapacity'] as int,
    isActive: json['isActive'] as bool,
  );
}
